### ISSUE TITLE
Wine-Wayland patch was just updated to 7.7.3

### ISSUE THREAD

**Issue by gel-crabs** [Frogging-Family/community-patches#113]
https://github.com/varmd/wine-wayland

**Comment by gel-crabs** [comment]
Actually the above repo isn't what the patch here uses, it's this one:

https://gitlab.collabora.com/alf/wine/-/tree/wayland

The repo is updated pretty frequently, perhaps there should be a separate option in wine-tkg-git for this. The Wayland patch here has been broken for a while now.

**Comment by Nolij** [comment]
Does anyone know how the existing patch was generated? It seems to have been generated by `git apply-patch`, but the best guess I have (`git apply-patch master..wayland --stdout`) generates 13604 patches, a few more than the 147 in the latest version. There's some sort of filtering being used but I'm not sure what it is. That or master is the wrong branch to track from.

**Comment by Tk-Glitch** [comment]
The wayland driver usefulness in itself is very debatable to begin with, at least for now and the next couple years. It's also kinda brittle regarding -devel updates so I'm not comfortable offering an option for it in wine-tkg-git, and that's why it's in community-patches. When it'll be ready for the show it'll land in mainline, even if only after a little time in staging.

> https://github.com/varmd/wine-wayland

That repo is pretty useless to us as you figured.
I have updated the patchset to current with https://github.com/Frogging-Family/community-patches/commit/65419d6f5dd12d82d6aa3ba7d81297738591e3c4

Closing

**Comment by Nolij** [comment]
> The wayland driver usefulness in itself is very debatable to begin with, at least for now and the next couple years. It's also kinda brittle regarding -devel updates so I'm not comfortable offering an option for it in wine-tkg-git, and that's why it's in community-patches. When it'll be ready for the show it'll land in mainline, even if only after a little time in staging.
> 
> > https://github.com/varmd/wine-wayland
> 
> That repo is pretty useless to us as you figured. I have updated the patchset to current with [65419d6](https://github.com/Frogging-Family/community-patches/commit/65419d6f5dd12d82d6aa3ba7d81297738591e3c4)
> 
> Closing

How did you generate the updated patch? That way those who want to use it can update it themselves.

**Comment by Tk-Glitch** [comment]
> How did you generate the updated patch? That way those who want to use it can update it themselves.

Since I am not aware of a way to do it on gitlab the way you can do it on github (directly downloading a diff from a comparison), I cloned the repo, found the wine 7.10 commit and generated a diff from that commit (631f4d5e42b7c2f705eb4b40eeea0ce1866bf513) to the HEAD. The commit id will change on each revision so that's not really useful :sweat_smile: 

Here's the comparison btw, if someone knows how to get a diff out of it from gitlab: https://gitlab.collabora.com/alf/wine/-/compare/631f4d5e42b7c2f705eb4b40eeea0ce1866bf513...wayland?from_project_id=2847

**Comment by Nolij** [comment]
Thanks. Trying to use this for 1 specific thing: some games (Roblox) fail to grab my cursor on Wayland, and debug logs are showing no differences. Figured Wayland driver might help. I don't know yet if it does or not considering it doesn't appear to be using the Wayland driver despite compiling it. That's for another issue though.

**Comment by Tk-Glitch** [comment]
I just updated the note about it as it requires special care: https://github.com/Frogging-Family/community-patches/commit/d30884e67d912635435603487d6e63f6059fc662
